### PR TITLE
DiscoveryEngineTargetSite: initial types

### PR DESCRIPTION
**Pull request by justinsb**
- **DiscoveryEngineTargetSite: initial types**
- **DiscoveryEngineTargetSite: generated apimachinery**

### CODE REVIEW

File: apis/discoveryengine/v1alpha1/targetsite_reference.go [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+	}
+	if actualExternalRef == "" {
+		return "", k8s.NewReferenceNotReadyError(u.GroupVersionKind(), key)
+	}
+	r.External = actualExternalRef
+	return r.External, nil
+}
+
+// NewTargetSiteLinkFromObject builds a TargetSiteLink from the Config Connector object.
+func NewTargetSiteLinkFromObject(ctx context.Context, reader client.Reader, obj *DiscoveryEngineDataStoreTargetSite) (*DiscoveryEngineDataStoreID, *TargetSiteLink, error) {
```
Comment (yuwenma) [inline apis/discoveryengine/v1alpha1/targetsite_reference.go:99]: Just curious, this is not called and also not from the SciFi template. Is it from Gemini?

File: apis/discoveryengine/v1alpha1/targetsite_types.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+// +kcc:proto=google.cloud.discoveryengine.v1.TargetSite
+type DiscoveryEngineDataStoreTargetSiteSpec struct {
+	// The DataStore this target site should be part of.
+	DataStoreRef *DiscoveryEngineDataStoreRef `json:"dataStoreRef,omitempty"`
+
+	// The resource ID is server-generated, so no ResourceID field
+
+	// Required. Input only. The user provided URI pattern from which the
+	// `generated_uri_pattern` is generated.
+	ProvidedUriPattern *string `json:"providedUriPattern,omitempty"`
```
Comment (yuwenma) [inline apis/discoveryengine/v1alpha1/targetsite_types.go:34]: nit: Uri --> URI

File: apis/discoveryengine/v1alpha1/targetsite_types.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+}
+
+// DiscoveryEngineDataStoreTargetSiteObservedState is the state of the DiscoveryEngineDataStoreTargetSite resource as most recently observed in GCP.
+// +kcc:proto=google.cloud.discoveryengine.v1.TargetSite
+type DiscoveryEngineDataStoreTargetSiteObservedState struct {
+	// Output only. This is system-generated based on the provided_uri.
+	GeneratedUriPattern *string `json:"generatedUriPattern,omitempty"`
+
+	// Output only. Root domain of the provided_uri.
+	RootDomainUri *string `json:"rootDomainUri,omitempty"`
```
Comment (yuwenma) [inline apis/discoveryengine/v1alpha1/targetsite_types.go:72]: nit: Uri --> URI

File: apis/discoveryengine/v1alpha1/targetsite_types.go [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *DiscoveryEngineDataStoreTargetSiteObservedState `json:"observedState,omitempty"`
+}
+
+// DiscoveryEngineDataStoreTargetSiteObservedState is the state of the DiscoveryEngineDataStoreTargetSite resource as most recently observed in GCP.
+// +kcc:proto=google.cloud.discoveryengine.v1.TargetSite
+type DiscoveryEngineDataStoreTargetSiteObservedState struct {
+	// Output only. This is system-generated based on the provided_uri.
+	GeneratedUriPattern *string `json:"generatedUriPattern,omitempty"`
```
Comment (yuwenma) [inline apis/discoveryengine/v1alpha1/targetsite_types.go:69]: ditto

File: apis/discoveryengine/v1alpha1/targetsite_types.go (hunk 2 above)
Comment (yuwenma) [reply apis/discoveryengine/v1alpha1/targetsite_types.go:34]: Do you mind adding this acronym to the CRD validation list? https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/tests/apichecks/crds_test.go#L195

**Review by yuwenma — COMMENTED** [review]
/approve

**Comment by justinsb** [comment]
Added URI to acronym validation and updated the json field tags.  I didn't update the go field names because the mapper uses those and is case sensitive (though I think elsewhere I added "near misses" to deal with this problem)

**Comment by yuwenma** [comment]
Looks great. Thank! Waiting for presubmit to pass

**Pull request merged**